lib/change_detection/dirty_checking_change_detector.dart

```diff
@@ -1538,12 +1546,8 @@ class DuplicateMap {
 bool _looseIdentical(dst, src) {
   if (identical(dst, src)) return true;
 
-  if (dst is String && src is String && dst == src) {
-    // this is false change in strings we need to recover, and pretend it is the same. We save the
-    // value so that next time identity can pass
-    dst = src;
-    return true;
-  }
+  // Dart could have string1 == string2 without string1 === string2
```

Comment:
Dart hasn't had `===` since '12 [http://stackoverflow.com/questions/13275506/what-is-the-difference-between-and-in-dart], I think you mean without `identical(dst, src)`.